src/Front/Scanner.x

```diff
@@ -114,20 +118,25 @@ tokens :-
     \&                  { keyword TAmpersand }
 
     -- | Other stringy tokens.
-    @operator           { strTok TOp }
-    \` @identifier \`   { strTok (TOp . dropEnds 1 1) }
-    @identifier         { strTok TId }
+    @operator           { strTok (TOp . fromString) }
+    \` @identifier \`   { strTok (TOp . fromString . dropEnds 1 1) }
+    @identifier         { strTok (TId . fromString) }
     $digit+             { strTok (TInteger . read) }
   }
 
 {
+-- | Internal compiler error for unreachable code.
+internalErr :: String -> Alex a
+internalErr s = alexError $ "_i:" ++ s
+
 -- | User-facing syntax error.
 syntaxErr :: String -> Alex a
-syntaxErr s = alexError $ "Syntax error: " ++ s
+syntaxErr s = alexError $ "_s:" ++ s
 
--- | Internal compiler error for unreachable code.
-internalErr :: String -> Alex a
-internalErr s = alexError $ "Internal error: " ++ s
+liftErr :: String -> Error
+liftErr ('_':'i':':':e) = UnexpectedError $ fromString e
```

Comment:
Within the alex monad, errors are thrown as strings, so we need to parse them out into `Compiler.Error`

Reply:
cursed but understandable :(